orchestra/proc-macro/src/impl_message_wrapper.rs

```diff
@@ -73,12 +73,30 @@ pub(crate) fn impl_message_wrapper_enum(info: &OrchestraInfo) -> Result<proc_mac
 			}
 		}
 
+		impl ::std::convert::From< #message_wrapper > for () {
+			fn from(message: #message_wrapper) -> Self {
+				match message {
+					#message_wrapper :: Empty => (),
+					_ => panic!("Message is not of type {}", stringify!(#message_wrapper)),
+				}
+			}
+		}
+
 		#(
 			impl ::std::convert::From< #consumes > for #message_wrapper {
 				fn from(message: #consumes) -> Self {
 					#message_wrapper :: #consumes_variant ( message )
 				}
 			}
+
+			impl ::std::convert::From< #message_wrapper > for #consumes {
+				fn from(message: #message_wrapper) -> Self {
+					match message {
+						#message_wrapper :: #consumes_variant ( inner ) => inner,
+						_ => panic!("Message is not of type {}", stringify!(#consumes)),
+					}
+				}
+			}
```

Comment:
Where do we need this?

Reply:
The result of `try_send` for an output channel is `#message_wrapper`, but we need to convert it back to `#consumes`. Removing that produces the following error:

```
error[E0277]: the trait bound `misc::Plinko: From<AllMessages>` is not satisfied
    --> /Users/vstakhov/orchestra/target/debug/build/expander-e4d93808c85fcd34/out/orchestra-expansion-e4d679c7c3ab.rs:95:25
     |
95   |       AwesomeSenderTrait, DuoSubsystemContext < MsgStrukt > :
     |  _________________________^
96   | |     AwesomeContextTrait, GoblinTower : crate :: Subsystem <
     | |_______________________^ the trait `From<AllMessages>` is not implemented for `misc::Plinko`
```